src/System.CommandLine/Option{T}.cs

```diff
@@ -57,8 +61,96 @@ public class Option<T> : Option, IValueDescriptor<T>
             string[] aliases,
             Func<T> defaultValueFactory,
             string? description = null)
-            : base(aliases, description, new Argument<T>(defaultValueFactory ?? throw new ArgumentNullException(nameof(defaultValueFactory))))
+            : this(aliases, description, new Argument<T>(defaultValueFactory ?? throw new ArgumentNullException(nameof(defaultValueFactory))))
+        {
+        }
+
+        private protected Option(
+            string name,
+            string? description,
+            Argument<T> argument)
+            : base(name, description)
+        {
+            argument.AddParent(this);
+            _argument = argument;
+        }
+
+        private protected Option(
+            string[] aliases,
+            string? description,
+            Argument<T> argument)
+            : base(aliases, description)
+        {
+            argument.AddParent(this);
+            _argument = argument;
+        }
+
+        internal sealed override Argument Argument => _argument;
```

Comment:
By making this property `sealed` we enforce all `Option<T>` derived types to always pass an instance of `Argument<T>` to the `ctor`. By doing that, we are always sure that  `_argument` (the field) and `Argument` (the property) always refer to the same instance of an object. In the past it was not always a thing?

Another benefit is that when the derived types are being created, only one `Argument<T>` is being allocated.